### PR TITLE
Add no_offset_view

### PR DESCRIPTION
**Pull request by tpapp**
Add a function to return a view with 1-based indexing. Cf [discussion](https://discourse.julialang.org/t/lift-and-wrap-array-with-custom-indexes/19436/11).

I thought this package would be the natural place for this, since it is implemented using `OffsetArray`, but feel free to reject if you don't think it fits into the scope of this package.

Suggestions about the name are welcome.

### CODE REVIEW

File: src/OffsetArrays.jl [hunk 1]
```diff
@@ -193,4 +193,39 @@ printindices(io::IO, ind1) = print(io, _unslice(ind1))
...
+
+julia> A
+2×3 Array{Int64,2}:
+ -9  3  5
+  2  4  6
+```
+"""
+function no_offset_view(A::AbstractArray)
+    if Base.has_offset_axes(A)
+        OffsetArray(A, 1 .+ size(A))
```
Comment (timholy) [inline src/OffsetArrays.jl]: This won't work in general:
```julia
julia> using CatIndices, OffsetArrays

julia> v = BidirectionalVector([1,2,3])
BidirectionalVector{Int64} with indices CatIndices.URange(1,3):
 1
 2
 3

julia> pushfirst!(v, 0)
BidirectionalVector{Int64} with indices CatIndices.URange(0,3):
 0
 1
 2
 3

julia> OffsetArrays.no_offset_view(v)
OffsetArray(::BidirectionalVector{Int64}, 5:8) with eltype Int64 with indices 5:8:
 0
 1
 2
 3
```

```suggestion
        OffsetArray(A, map(r->1-first(r), axes(A)))
```

Comment (tpapp) [reply src/OffsetArrays.jl]: Thanks, I will add a test for this.

**Comment by timholy** [comment]
(If you need to rebase there is no need to worry about "credit" for me on that tiny change, just do whatever is easiest.)

**Comment by tpapp** [comment]
Thanks, just resolved the conflict.

**Comment by timholy** [comment]
Thanks, very nice work!

**Pull request merged**